src/3d/qgs3dmapexportsettings.h

```diff
@@ -0,0 +1,86 @@
+/***************************************************************************
+  qgs3dmapexportsettings.h
+  --------------------------------------
+  Date                 : July 2020
+  Copyright            : (C) 2020 by Belgacem Nedjima
+  Email                : gb underscore nedjima at esi dot dz
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGS3DMAPEXPORTSETTINGS_H
+#define QGS3DMAPEXPORTSETTINGS_H
+
+#include "qgis_3d.h"
+
+#include <QString>
+#include <QObject>
+#include <QDir>
+
+/**
+ * \brief The Qgs3DMapExportSettings class
+ * Manages the various settings the user can choose from when exorting a 3D scene
+ * \ingroup 3d
+ * \since QGIS 3.16
+ */
+class _3D_EXPORT Qgs3DMapExportSettings : public QObject
+{
+    Q_OBJECT
```

Comment:
This shouldn't be a QObject either (unless I'm missing something?)

Reply:
I made the Qgs3DMapExportSettings behave as a parent of  Qgs3DExportObject so that every Qgs3DExportObject instantiated using a Qgs3DMapExportSettings will be deallocated once the parent Qgs3DMapExportSettings is deleted.

Reply:
That's using a sledgehammer approach :laugh: If you're not using signals/slots/some complex shared ownership, it's much clearer to just use plain old pointers and clean them up in the destructor (or, the modern approach of a std::vector containing std::unique_ptr, so that they'll be cleaned up automatically)

Reply:
Yes this class shouldn't be QObject. It is only necessary for the Qgs3DSceneExporter to inherit from Qt3DCore::QEntity so that every QEntity we create dynamically will be deallocated properly